### ISSUE TITLE
Webpack warnings when using monaco-textmate

### ISSUE THREAD

**Issue by epatpol** [NeekSandhu/monaco-textmate#2]
```
@theia/example-browser: WARNING in /home/epatpol/Projects/Github/theia/node_modules/monaco-textmate/dist/grammar.js
@theia/example-browser: (Emitted value instead of an instance of Error) Cannot find source file '../src/grammar.ts': Error: Can't resolve '../src/grammar.ts' in '/home/epatpol/Projects/Github/theia/node_modules/monaco-textmate/dist'
@theia/example-browser:  @ /home/epatpol/Projects/Github/theia/node_modules/monaco-textmate/dist/main.js 17:18-38
@theia/example-browser:  @ /home/epatpol/Projects/Github/theia/packages/monaco-textmate/lib/browser/monaco-textmate-service.js
@theia/example-browser:  @ /home/epatpol/Projects/Github/theia/packages/monaco-textmate/lib/browser/monaco-textmate-frontend-module.js
@theia/example-browser:  @ ./src-gen/frontend/index.js
@theia/example-browser: WARNING in /home/epatpol/Projects/Github/theia/node_modules/monaco-textmate/dist/grammarReader.js
@theia/example-browser: (Emitted value instead of an instance of Error) Cannot find source file '../src/grammarReader.ts': Error: Can't resolve '../src/grammarReader.ts' in '/home/epatpol/Projects/Github/theia/node_modules/monaco-textmate/dist'
@theia/example-browser:  @ /home/epatpol/Projects/Github/theia/node_modules/monaco-textmate/dist/main.js 15:24-50
@theia/example-browser:  @ /home/epatpol/Projects/Github/theia/packages/monaco-textmate/lib/browser/monaco-textmate-service.js
@theia/example-browser:  @ /home/epatpol/Projects/Github/theia/packages/monaco-textmate/lib/browser/monaco-textmate-frontend-module.js
@theia/example-browser:  @ ./src-gen/frontend/index.js
@theia/example-browser: WARNING in /home/epatpol/Projects/Github/theia/node_modules/onigasm/lib/index.js
@theia/example-browser: (Emitted value instead of an instance of Error) Cannot find source file '../src/index.ts': Error: Can't resolve '../src/index.ts' in '/home/epatpol/Projects/Github/theia/node_modules/onigasm/lib'
@theia/example-browser:  @ /home/epatpol/Projects/Github/theia/packages/monaco-textmate/lib/browser/monaco-textmate-frontend-contribution.js 55:16-34
@theia/example-browser:  @ /home/epatpol/Projects/Github/theia/packages/monaco-textmate/lib/browser/monaco-textmate-frontend-module.js
@theia/example-browser:  @ ./src-gen/frontend/index.js
@theia/example-browser: WARNING in /home/epatpol/Projects/Github/theia/node_modules/monaco-textmate/dist/json.js
@theia/example-browser: (Emitted value instead of an instance of Error) Cannot find source file '../src/json.ts': Error: Can't resolve '../src/json.ts' in '/home/epatpol/Projects/Github/theia/node_modules/monaco-textmate/dist'
@theia/example-browser:  @ /home/epatpol/Projects/Github/theia/node_modules/monaco-textmate/dist/grammarReader.js 8:15-32
@theia/example-browser:  @ /home/epatpol/Projects/Github/theia/node_modules/monaco-textmate/dist/main.js
@theia/example-browser:  @ /home/epatpol/Projects/Github/theia/packages/monaco-textmate/lib/browser/monaco-textmate-service.js
@theia/example-browser:  @ /home/epatpol/Projects/Github/theia/packages/monaco-textmate/lib/browser/monaco-textmate-frontend-module.js
@theia/example-browser:  @ ./src-gen/frontend/index.js
@theia/example-browser: WARNING in /home/epatpol/Projects/Github/theia/node_modules/monaco-textmate/dist/main.js
@theia/example-browser: (Emitted value instead of an instance of Error) Cannot find source file '../src/main.ts': Error: Can't resolve '../src/main.ts' in '/home/epatpol/Projects/Github/theia/node_modules/monaco-textmate/dist'
@theia/example-browser:  @ /home/epatpol/Projects/Github/theia/packages/monaco-textmate/lib/browser/monaco-textmate-service.js 55:24-50
@theia/example-browser:  @ /home/epatpol/Projects/Github/theia/packages/monaco-textmate/lib/browser/monaco-textmate-frontend-module.js
@theia/example-browser:  @ ./src-gen/frontend/index.js
@theia/example-browser: WARNING in /home/epatpol/Projects/Github/theia/node_modules/monaco-textmate/dist/matcher.js
@theia/example-browser: (Emitted value instead of an instance of Error) Cannot find source file '../src/matcher.ts': Error: Can't resolve '../src/matcher.ts' in '/home/epatpol/Projects/Github/theia/node_modules/monaco-textmate/dist'
@theia/example-browser:  @ /home/epatpol/Projects/Github/theia/node_modules/monaco-textmate/dist/grammar.js 8:18-38
@theia/example-browser:  @ /home/epatpol/Projects/Github/theia/node_modules/monaco-textmate/dist/main.js
@theia/example-browser:  @ /home/epatpol/Projects/Github/theia/packages/monaco-textmate/lib/browser/monaco-textmate-service.js
@theia/example-browser:  @ /home/epatpol/Projects/Github/theia/packages/monaco-textmate/lib/browser/monaco-textmate-frontend-module.js
@theia/example-browser:  @ ./src-gen/frontend/index.js
@theia/example-browser: WARNING in /home/epatpol/Projects/Github/theia/node_modules/onigasm/lib/onigasmH.js
@theia/example-browser: (Emitted value instead of an instance of Error) Cannot find source file '../src/onigasmH.ts': Error: Can't resolve '../src/onigasmH.ts' in '/home/epatpol/Projects/Github/theia/node_modules/onigasm/lib'
@theia/example-browser:  @ /home/epatpol/Projects/Github/theia/node_modules/onigasm/lib/index.js 3:19-40
@theia/example-browser:  @ /home/epatpol/Projects/Github/theia/packages/monaco-textmate/lib/browser/monaco-textmate-frontend-contribution.js
@theia/example-browser:  @ /home/epatpol/Projects/Github/theia/packages/monaco-textmate/lib/browser/monaco-textmate-frontend-module.js
@theia/example-browser:  @ ./src-gen/frontend/index.js
@theia/example-browser: WARNING in /home/epatpol/Projects/Github/theia/node_modules/monaco-textmate/dist/registry.js
@theia/example-browser: (Emitted value instead of an instance of Error) Cannot find source file '../src/registry.ts': Error: Can't resolve '../src/registry.ts' in '/home/epatpol/Projects/Github/theia/node_modules/monaco-textmate/dist'
@theia/example-browser:  @ /home/epatpol/Projects/Github/theia/node_modules/monaco-textmate/dist/main.js 14:19-40
@theia/example-browser:  @ /home/epatpol/Projects/Github/theia/packages/monaco-textmate/lib/browser/monaco-textmate-service.js
@theia/example-browser:  @ /home/epatpol/Projects/Github/theia/packages/monaco-textmate/lib/browser/monaco-textmate-frontend-module.js
@theia/example-browser:  @ ./src-gen/frontend/index.js
@theia/example-browser: WARNING in /home/epatpol/Projects/Github/theia/node_modules/monaco-textmate/dist/rule.js
@theia/example-browser: (Emitted value instead of an instance of Error) Cannot find source file '../src/rule.ts': Error: Can't resolve '../src/rule.ts' in '/home/epatpol/Projects/Github/theia/node_modules/monaco-textmate/dist'
@theia/example-browser:  @ /home/epatpol/Projects/Github/theia/node_modules/monaco-textmate/dist/grammar.js 7:15-32
@theia/example-browser:  @ /home/epatpol/Projects/Github/theia/node_modules/monaco-textmate/dist/main.js
@theia/example-browser:  @ /home/epatpol/Projects/Github/theia/packages/monaco-textmate/lib/browser/monaco-textmate-service.js
@theia/example-browser:  @ /home/epatpol/Projects/Github/theia/packages/monaco-textmate/lib/browser/monaco-textmate-frontend-module.js
@theia/example-browser:  @ ./src-gen/frontend/index.js
@theia/example-browser: WARNING in /home/epatpol/Projects/Github/theia/node_modules/monaco-textmate/dist/theme.js
@theia/example-browser: (Emitted value instead of an instance of Error) Cannot find source file '../src/theme.ts': Error: Can't resolve '../src/theme.ts' in '/home/epatpol/Projects/Github/theia/node_modules/monaco-textmate/dist'
@theia/example-browser:  @ /home/epatpol/Projects/Github/theia/node_modules/monaco-textmate/dist/main.js 16:16-34
@theia/example-browser:  @ /home/epatpol/Projects/Github/theia/packages/monaco-textmate/lib/browser/monaco-textmate-service.js
@theia/example-browser:  @ /home/epatpol/Projects/Github/theia/packages/monaco-textmate/lib/browser/monaco-textmate-frontend-module.js
@theia/example-browser:  @ ./src-gen/frontend/index.js
@theia/example-browser: WARNING in /home/epatpol/Projects/Github/theia/node_modules/monaco-textmate/dist/utils.js
@theia/example-browser: (Emitted value instead of an instance of Error) Cannot find source file '../src/utils.ts': Error: Can't resolve '../src/utils.ts' in '/home/epatpol/Projects/Github/theia/node_modules/monaco-textmate/dist'
@theia/example-browser:  @ /home/epatpol/Projects/Github/theia/node_modules/monaco-textmate/dist/grammar.js 6:16-34
@theia/example-browser:  @ /home/epatpol/Projects/Github/theia/node_modules/monaco-textmate/dist/main.js
@theia/example-browser:  @ /home/epatpol/Projects/Github/theia/packages/monaco-textmate/lib/browser/monaco-textmate-service.js
@theia/example-browser:  @ /home/epatpol/Projects/Github/theia/packages/monaco-textmate/lib/browser/monaco-textmate-frontend-module.js
@theia/example-browser:  @ ./src-gen/frontend/index.js
```

I think this has something to do with the sourcemaps but I'm not sure how to fix it. Thing is I was trying to debug the frontend from within chrome/firefox and I couldn't access the monaco-textmate source files, which is a bit troublesome when debugging. Any ideas/workarounds to have sourcemap working when using webpack?

**Comment by zikaari** [comment]
It's very likely that `webpack.config.js` is telling Webpack to do funny stuff. Can you share your config? Could be resolving rules or something.

It works fine on my system, in fact try monaco-textmate in an isolated project (of course with Webpack) it shouldn't complain.

**Comment by epatpol** [comment]
Here's my current webpack config https://gist.github.com/epatpol/7182766dfe7edd133fd0bb13d49a6047

**Comment by zikaari** [comment]
~~Regarding sourcemaps, please note that they are generated and available inside npm distribution. See inside `node_modules/monaco-textmate/dist` and you'll find map files. It's just that Webpack doesn't pick them up and consolidate them with output file.~~

~~Maybe create an issue in Webpack repo to have it looked over? Thanks~~

Edit: I didn't know about existence of source-map-loader. I'll need to investigate that guy.

**Comment by zikaari** [comment]
The error is caused by the fact that `src/` is in `.npmignore` so it doesn't get shipped when you `npm install monaco-textmate`. Therefore `source-map-loader` fails to do its job.

I'm gonna fix that too later today.

**Comment by zikaari** [comment]
Fixed in `2.0.0`. Please run `npm i monaco-textmate` again to update.
- https://github.com/NeekSandhu/monaco-textmate/commit/68213d2c162adb4e5efb7a3153bb4efb1e58142e - unlist src/ from .npmignore for sourcemap's sake in prod

**Comment by epatpol** [comment]
Thank you! Now do you think it should/could also be done with `onigasm`? Seeing as it's a peer dependency to this project this one will also get source map warnings if it follows the same project structure. For now I'll close this one.
